### PR TITLE
chore: add Job functions to deploy package

### PR DESCRIPTION
**Pull request by bvtujo**
This is a prerequisite PR for the ongoing `job logs` work. This adds two methods to `deploy` to allow Copilot to check whether a given job is deployed in an environment, and to list all deployed jobs by app and env. 

This PR needs to be merged before I can finish work on `job logs Ask`.

Related #2272 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1212,6 +1212,7 @@ golang.org/x/tools v0.0.0-20200501065659-ab2804fb9c9d/go.mod h1:EkVYQZoAsY45+roY
 golang.org/x/tools v0.0.0-20200502202811-ed308ab3e770/go.mod h1:EkVYQZoAsY45+roYkvgYkIh4xh/qjgUK9TdY2XT94GE=
 golang.org/x/tools v0.0.0-20200512131952-2bc93b1c0c88/go.mod h1:EkVYQZoAsY45+roYkvgYkIh4xh/qjgUK9TdY2XT94GE=
 golang.org/x/tools v0.0.0-20200804011535-6c149bb5ef0d/go.mod h1:njjCfa9FT2d7l9Bc6FUM5FLjQPp3cFF28FI3qnDFljA=
+golang.org/x/tools v0.0.0-20201224043029-2b0845dc783e h1:4nW4NLDYnU28ojHaHO8OVxFHk/aQ33U01a9cjED+pzE=
```
Comment (iamhopaul123) [inline go.sum:1220]: Do we want this change?

**Review by iamhopaul123 — APPROVED** [review]
LGTM ship it 🚢

File: go.sum (hunk 1 above)
Comment (bvtujo) [reply go.sum:1220]: Go modules seems to want it; it remains after `go mod tidy` 😐

File: internal/pkg/deploy/deploy.go [hunk 2]
```diff
@@ -71,32 +77,50 @@ func NewStore(store ConfigStoreClient) (*Store, error) {
...
 	for ind, resource := range resources {
-		svcName := resource.Tags[ServiceTagKey]
-		if svcName == "" {
-			return nil, fmt.Errorf("service with ARN %s is not tagged with %s", resource.ARN, ServiceTagKey)
+		name := resource.Tags[ServiceTagKey]
+		if name == "" {
+			return nil, fmt.Errorf("%s resource with ARN %s is not tagged with %s", workloadType, resource.ARN, ServiceTagKey)
 		}
-		svc, err := s.configStore.GetService(appName, svcName)
+		wkld, err := s.configStore.GetService(appName, name)
```
Comment (Lou1415926) [inline internal/pkg/deploy/deploy.go]: Should we use `configStore.GetJob` for job? Or `configStore.GetWorkload`.

**Review by Lou1415926 — COMMENTED** [review]
LGTM! One question ⬇️

File: internal/pkg/deploy/deploy.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/deploy/deploy.go]: Good call! I missed this one. We should probably determine by what workload type it is.

**Review by Lou1415926 — APPROVED** [review]
:shipit: !

**Pull request merged**